libs/utils/utils.js

```diff
@@ -923,10 +923,10 @@ function decorateMeta() {
   });
 }
 
-function decorateDocumentExtras(config) {
+async function decorateDocumentExtras(config) {
   decorateMeta();
   decorateHeader();
-  decorateFooterPromo(config);
+  await decorateFooterPromo(config);
```

Comment:
Since this is usually pushed out to the button of the page, does it really make sense to await this?

Reply:
Yeah it needs to finish before the fragments are decorated. (You'll notice that the outcome of `decorateFooterPromo` is that it adds a fragment link onto the page). If you think it's a problem, I'll need to do something custom to load the fragment and avoid a race condition

Reply:
This runs before the LCP.
Doing something custom is worth it IMO, as is happening in the most crucial period of the whole rendering setup.

Reply:
@auniverseaway @chrischrischris @rgclayton Thoughts on this?

Reply:
This comes with the same performance penalty as a regular fragment, correct? Meaning, this would be a known use case.

One potential thought, since this is loaded at the bottom of the page, would be to use [loadDeferred](https://github.com/adobecom/milo/blob/main/libs/utils/utils.js#L864). That means your fragments are all ready decorated so another call would need to be made. 

Another thought is taking a similar approach as inline fragments and using  [processSection](https://github.com/adobecom/milo/blob/main/libs/utils/utils.js#L973C16-L973C30)

Reply:
A difference is that we now also load the `taxonomy.json` before LCP

Reply:
@mokimo @rgclayton Thanks for the feedback! I talked with @chrischrischris and I think we've gotten to a good solution. I moved `decorateFooterPromo` into `documentPostSectionLoading` so it won't block LCP. Then I'm importing `fragment.js` and loading the fragment.

Please take another look and let me know what you think

Reply:
Perfect! 👍 Thanks for following up on this - looking great